test/acceptance/mock_test.js

```diff
@@ -8,63 +8,57 @@ const fetchUsers = response => response.url() === 'https://jsonplaceholder.typic
 
 Scenario('change statusCode @Puppeteer', (I) => {
   I.amOnPage('/form/fetch_call');
-  I.click('GET POSTS');
-  I.waitForResponse(fetchPost, 3);
   I.mockRequest('GET', 'https://jsonplaceholder.typicode.com/*', 404);
   I.click('GET POSTS');
-  I.see('Can not load data!');
+  I.waitForText('Can not load data!', 1, '#data');
   I.stopMocking();
 });
 
 Scenario('change response data @Puppeteer', (I) => {
   I.amOnPage('/form/fetch_call');
-  I.click('GET COMMENTS');
-  I.waitForResponse(fetchComments, 3);
   I.mockRequest('GET', 'https://jsonplaceholder.typicode.com/*', {
     modified: 'This is modified from mocking',
   });
   I.click('GET COMMENTS');
-  I.see('This is modified from mocking', '#data');
+  I.waitForText('This is modified from mocking', 1, '#data');
   I.stopMocking();
 });
 
 Scenario('change response data for multiple requests @Puppeteer', (I) => {
   I.amOnPage('/form/fetch_call');
-  I.click('GET USERS');
-  I.waitForResponse(fetchUsers, 3);
   I.mockRequest(
     'GET',
     [
       'https://jsonplaceholder.typicode.com/posts/*',
       'https://jsonplaceholder.typicode.com/comments/*',
-      'https://jsonplaceholder.typicode.com/users/*',
     ],
     {
       modified: 'MY CUSTOM DATA',
     },
   );
   I.click('GET POSTS');
-  I.see('MY CUSTOM DATA', '#data');
+  I.waitForText('MY CUSTOM DATA', 1, '#data');
   I.click('GET COMMENTS');
-  I.see('MY CUSTOM DATA', '#data');
+  I.waitForText('MY CUSTOM DATA', 1, '#data');
   I.click('GET USERS');
-  I.see('MY CUSTOM DATA', '#data');
+  pause();
+  I.dontSee('MY CUSTOM DATA', '#data');
   I.stopMocking();
 });
 
 Scenario(
   'should request for original data after mocking stopped @Puppeteer',
   (I) => {
     I.amOnPage('/form/fetch_call');
-    I.click('GET COMMENTS');
     I.mockRequest('GET', 'https://jsonplaceholder.typicode.com/*', {
-      comment: 'CUSTOM',
+      comment: 'CUSTOM _uniqueId_u4805sd23',
```

Comment:
Just to make sure that `CUSTOM` is not from actual data but from mocked data.

Reply:
Then better to generate unique id